xui/additional_files_unzipped/s3_upload_object/s3_upload_object/cb_plugin_214_HXw030a_KRV1EVv.py

```diff
@@ -0,0 +1,42 @@
+import os
+from django.shortcuts import get_object_or_404
+from resourcehandlers.aws.models import AWSHandler
+from resources.models import Resource
+from utilities.logger import ThreadLogger
+from utilities.get_current_userprofile import get_current_userprofile
+
+# Global Variables
+logger = ThreadLogger(__name__)
+
+def inbound_web_hook_post(*args, parameters={}, **kwargs):
+    """
+    Upload a single item
+        :param resource_id: The ID of the s3 storage bucket resource
+        :param bucket_name: The name of the s3 storage bucket
+        :param path: The current location for the item to upload to
+        :param file_name: The name of the file being uploaded
+        :param object_file: The file data
+    """
+    try:
+        resource_id =  parameters.get("resource_id")
+        user = get_current_userprofile()
+        print(resource_id)
+        resource = get_object_or_404(Resource, pk=resource_id)
+        file_name = os.path.join(
+            parameters.get("path", None), parameters.get("file_name", None)
+        )
+        bucket_name = parameters.get("bucket_name", None)
+        file = parameters.get("object_file", None)
+
+        aws = get_object_or_404(AWSHandler, pk=resource.aws_rh_id)
+        s3_client = aws.get_boto3_client(None, "s3")
+        s3_client.put_object(Body=file, Bucket=bucket_name, Key=file_name)
+        
+        # Log this action
+        logger.info("User %s uploaded file %s to S3 bucket: %s" % (user, file_name, resource.name))
+        return {"status": True, "message": "Successfully Uploaded"}
+
+    except Exception as e:
+        error_message = e.args[0]
```

Comment:
one thing I recommend anytime our code blanket catches all exceptions: 
logger.exception("what it was attempting when it failed")

That will print the full TB (to the application.log file on prod CBs), which could be extremely useful in our t-shooting. I think the full TB is lost in the current exception handling.

https://docs.python.org/3/library/logging.html#logging.Logger.exception